sound/soc/codecs/rt715-sdca.c

```diff
@@ -0,0 +1,943 @@
+// SPDX-License-Identifier: GPL-2.0
+// 
+// rt715-sdca.c -- rt715 ALSA SoC audio driver
+// 
+// Copyright(c) 2019 Realtek Semiconductor Corp.
+// 
+// ALC715 ASoC Codec Driver based Intel Dummy SdW codec driver
+// 
+// 
+
+#include <linux/module.h>
+#include <linux/moduleparam.h>
+#include <linux/version.h>
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/delay.h>
+#include <linux/i2c.h>
+#include <linux/pm_runtime.h>
+#include <linux/pm.h>
+#include <linux/soundwire/sdw.h>
+#include <linux/gpio.h>
+#include <linux/regmap.h>
+#include <linux/slab.h>
+#include <linux/platform_device.h>
+#include <linux/regulator/consumer.h>
+#include <linux/gpio/consumer.h>
+#include <linux/of.h>
+#include <linux/of_gpio.h>
+#include <linux/of_device.h>
+#include <sound/core.h>
+#include <sound/pcm.h>
+#include <sound/pcm_params.h>
+#include <sound/soc.h>
+#include <sound/soc-dapm.h>
+#include <sound/initval.h>
+#include <sound/tlv.h>
+#include <sound/hda_verbs.h>
```

Comment:
I don't see any use of gpio, i2c, regulator or of_ headers? also delay.h, hda_verbs.h? Please, verify, that all headers are really needed.